node/src/actors/json_rpc/json_rpc_methods.rs

```diff
@@ -1180,6 +1184,28 @@ pub async fn get_balance(params: Params) -> JsonRpcResult {
         .await
 }
 
+/// Get supply info
+pub async fn get_supply_info() -> JsonRpcResult {
+    let chain_manager_addr = ChainManager::from_registry();
+
+    chain_manager_addr
+        .send(GetSupplyInfo)
+        .map(|res| {
+            res.map_err(internal_error)
+                .and_then(|supply_info| match supply_info {
+                    Ok(x) => match serde_json::to_value(&x) {
+                        Ok(x) => Ok(x),
+                        Err(e) => {
+                            let err = internal_error_s(e);
+                            Err(err)
+                        }
+                    },
+                    Err(e) => Err(internal_error_s(e)),
+                })
```

Comment:
All this block is ugly AF, but I hope this makes things a little more readable:
```suggestion
                .and_then(|supply_info| match supply_info {
                    Ok(x) => serde_json::to_value(&x).map_err(internal_error_s),
                    Err(e) => Err(internal_error_s(e)),
                })
```

Or maybe this could work:
```suggestion
                .and_then(|supply_info| supply_info
                    .and_then(|x| serde_json::to_value(&x).map_err(internal_error_s))
                    .map_err(internal_error_s)
                )
```

Reply:
This is copy-pasted from the other JSON-RPC handlers, the solution is to abstract this logic into a function and just do something like

```suggestion
                .and_then(|supply_info| to_json_result(supply_info))
```

Reply:
Loving `.and_then(to_json_result)` :stuck_out_tongue:

Reply:
But in a different PR